bake/bake.go

```diff
@@ -1303,6 +1304,16 @@ func updateContext(t *build.Inputs, inp *Input) {
 	if build.IsRemoteURL(t.ContextPath) {
 		return
 	}
+
+	// If the target includes a BUILDKIT_CONTEXT_KEEP_GIT_DIR=1 build argument,
+	// do not reuse the bake context state as it never keeps the .git directory
+	// (see ReadRemoteFiles).
+	if v, ok := target.Args["BUILDKIT_CONTEXT_KEEP_GIT_DIR"]; ok && v != nil {
```

Comment:
Not sure I get this. Shouldn't this only run in the `t.ContextPath == "." ` case above?